networkx/algorithms/traversal/breadth_first_search.py

```diff
@@ -418,3 +419,49 @@ def descendants_at_distance(G, source, distance):
         current_distance += 1
 
     return current_layer
+
+
+@not_implemented_for("directed")
+def _bfs_with_marks(G, start_node, check_set):
```

Comment:
I have a weak preference to keep the `_bfs_with_marks` function close to the code that uses it -- until we run into another problem that might want to stop the BFS at specific nodes.  Putting it here removes the context in which it is makes sense. Do other maintainers have thoughts on this? I might be persuaded.

Reply:
Makes sense. Moved back.